### ISSUE TITLE
Windows 10 | Node.js -  no output when using hamsters.js + webworker-threads package

### ISSUE THREAD

**Issue by infFenrir** [austinksmith/Hamsters.js#59]
```
Windows: 10
Node: v10.14.2
hamsters.js: 5.1.2
webworker-threads: 0.7.17
```
```
const Worker = require('webworker-threads').Worker
const hamsters = require('hamsters.js')

hamsters.init({
  Worker: Worker
})

const params = {
  array: [1, 2, 3, 4],
  threads: 1,
  aggregate: true
}

hamsters.promise(params, () => {
  for (let i=0; i<params.array.length; i++) {
    rtn.data.push(params.array[i] * 2)
  }
}).then( (results) => {
  console.log(results)
}).catch( (error) => {
  console.error(error)
})
```
Console -> Hamsters.js v5.1.2 Info: Hamsters.js v5.1.2 initialized using up to 4 threads. @ 1545047704894
and freezes...
What am i doing wrong?

**Comment by austinksmith** [comment]
It's working perfectly for me..

```js
hamsters.init({
  Worker: Worker
});

const params = {
  array: [1, 2, 3, 4],
  threads: 1,
  aggregate: true
}

hamsters.promise(params, () => {
  for (let i=0; i<params.array.length; i++) {
    rtn.data.push(params.array[i] * 2)
  }
}).then( (results) => {
  console.log(results)
}).catch( (error) => {
  console.error(error)
});
````

**Comment by austinksmith** [comment]
![screenshot from 2018-12-17 08-04-52](https://user-images.githubusercontent.com/2227747/50089048-ba305380-01d2-11e9-80ed-4ee1f6a7503f.png)
![screenshot from 2018-12-17 08-06-36](https://user-images.githubusercontent.com/2227747/50089049-ba305380-01d2-11e9-8998-6f3360d81989.png)

**Comment by InfSpell** [comment]
I do not know what is happening, they work separately but not together, and no error!...
Could you recommend another worker, or it's time to throw Windows?

**Comment by austinksmith** [comment]
Weird, so I went ahead and setup node on windows 10 and did some testing. I was able to get it to return an output the first time I ran it, but now It just hangs like you said. No error produced and the problem doesn't appear when using Ubuntu or a unix shell environment. 

This is a weird one and is unlikely to have anything to do with Hamsters.js itself as the logic works fine on the web and in node which leads me to believe there is something unique about how windows 10 is running node that is causing problems with promises resolving? Idk this one will require more research and is probably outside the scope of what I can change on my end but I'll dig into it a little more when I get some more time.

**Comment by austinksmith** [comment]
I have an idea on what might be going on but it's very odd that it would only happen on windows 10 and no where else

* Sending data and executing the work in a thread confirmed working
* Getting a response from a thread is confirmed working
* Running legacy mode is confirmed working
* Getting final output data is confirmed working

Where things break down is when calling ```resolve(output);``` just seems to fail silently, though if I forcibly call that earlier on in the process with a dummy output object it works so the promise isn't resolving properly. The library uses a promise internally so when you call hamsters.promise or hamsters.run ultimately a promise is invoked internally and then depending on how you called the library you can use a callback or a promise.then but either option is useless if the internal promise never resolves...

**Comment by InfSpell** [comment]
Thank you very much for your answer and tests, I thought that I was going crazy )))

**Comment by austinksmith** [comment]
I've referenced this issue here https://github.com/nodejs/node/issues/25156 maybe NodeJS team can provide a little insight.

**Comment by InfSpell** [comment]
Thanks, I will definitely see.

**Comment by austinksmith** [comment]
Hey @InfSpell can you do me a favor and test out the dev branch copy? I made some tweaks to how the library works internally and I think I may have figured out what the issue was though I'm not 100% certain just yet.

**Comment by InfSpell** [comment]
Hey @austinksmith. No, it does not work :(

Windows: 10
Node: v10.15.0
webworker-threads: 0.7.17

```
const Worker = require('webworker-threads').Worker
const hamsters = require('./Hamsters.js-dev/build/hamsters.node.min')

hamsters.init({
  Worker: Worker
})

function foo() {
  let params = {
    array: [0, 1, 2, 3, 4],
    threads: 1,
    aggregate: true
  }

  hamsters.run(params, function() {
      let arr = params.array
      arr.forEach(function(item) {
        rtn.data.push(item * 2)
      })
  }, function(results) {
    console.log(results)
  })
}

foo()
```
Console -> Hamsters.js v5.1.3 Info: Initialized using up to 4 threads. @ 1549908397515 
and freezes...
And for some reason your website stopped working ...

**Comment by austinksmith** [comment]
Weird, i fixed the website i'm gonna take another look at whats going on with this tonight and i'll report back tomorrow to see if it's fixed or not. It's frustrating because it works on linux and the web version works fine

**Comment by austinksmith** [comment]
https://github.com/austinksmith/Hamsters.js/issues/67

**Comment by austinksmith** [comment]
Hey there @infFenrir I know it's been a long time since you first created this issue ticket. However I have released a new Hamsters version that should resolve your issues and also adds support for Node.js native worker implementation. Please take a look at https://github.com/austinksmith/Hamsters.js/releases/tag/v5.1.3 and also read the updated wiki @ https://www.hamsters.io/wiki

Thank you for your patience.

**Comment by austinksmith** [comment]
I am marking this as resolved, I've just published v5.3.1 and tested it on Node.js both legacy mode and using Node.js native worker threads and all is working as expected https://github.com/austinksmith/Hamsters.js/releases/tag/v5.3.1
